src/zarr/group.py

```diff
@@ -248,16 +263,42 @@ def attrs(self):
     def info(self):
         return self.metadata.info
 
-    async def create_group(self, path: str, **kwargs) -> AsyncGroup:
+    async def create_group(
+        self, path: str, exists_ok: bool = False, attributes: dict[str, Any] = {}
+    ) -> AsyncGroup:
         return await type(self).create(
             self.store_path / path,
-            **kwargs,
+            attributes=attributes,
+            exists_ok=exists_ok,
+            zarr_format=self.metadata.zarr_format,
         )
 
-    async def create_array(self, path: str, **kwargs) -> AsyncArray:
+    async def create_array(
```

Comment:
`group.create_group` and `group.create_array` should preserve the zarr format of the `group`, which means we can't accept `**kwargs` in these methods without checking the contents of **kwargs for the absence of `zarr_format`. Since I'm not a huge fan of `**kwargs` anyway, I just elevated the underlying function signatures to this method, sans `zarr_format`, which is set automatically based on the zarr format of the `group`.

Reply:
We need to redo that in #1857